sceval_frontend/src/containers/SensorModule.tsx

```diff
@@ -123,132 +123,133 @@ export const SensorModule: React.FC<SensorModuleProps> = (props: SensorModulePro
-            setMounted(true);
-            // get selected device and module
-            const gateway = sessionStorage.getItem('selectedGateway');
-            const sensorModule = sessionStorage.getItem('selectedModule');
-            setSelectedModule(sensorModule);
-            setSelectedGateway(gateway);
-            if (gateway !== null) {
-                setTimeout(
-                    () => {
-                        // list sensor modules
-                        ModeConnection.listSensorModules(gateway);
-                    },
-                    1000
-                );
-            }
-            if (sensorModule) {
-                // for now, setting sensor module name to ID
-                setSensorModuleName(sensorModule);
-                // fetch module data from KV store
-                const deviceURL = MODE_API_BASE_URL + 'devices/' + gateway + '/kv/sensorModule' + sensorModule;
-                modeAPI.request('GET', deviceURL, {})
-                .then((response: any) => {
-                    setSelectedSensorModuleObj(response.data);
-                    
-                    const moduleSensors = response.data.value.sensors;
-                    // set name of sensor
-                    setSensorModuleName(response.data.value.name);
-                    // set full sensor list and quantity
-                    setFullSensorList(moduleSensors);
-                    setActiveSensorQuantity(moduleSensors.length);
-                    // determine offline sensors
-                    let sensorsOffline: any = fullALPsList.filter((sensor: any, index: any): boolean => {
-                        return !response.data.value.sensors.includes(sensor);
-                    });
-                    setOfflineSensors(sensorsOffline);
-                    // getTSDB seriesID for the particular module 
-                    const TSDBURL = `${MODE_API_BASE_URL}homes/${homeID}/smartModules/tsdb/timeSeries`;
-                    modeAPI.request('GET', TSDBURL, {})
-                    .then((tsdbResponse: any) => {  
-                        // filter response initially by selected module
-                        const filteredTSDBData: any = tsdbResponse.data.filter((tsdbData: any): boolean => {
-                            return tsdbData.id.includes(selectedModule);
-                        });
-                        // filter again for online sensors
-                        const onlineTSDBData: any = filteredTSDBData.filter((filteredData: any): boolean => {
-                            const sensorType = filteredData.id.split('-')[1].toUpperCase();
-                            return moduleSensors.includes(sensorType) && 
-                            !sensorType.includes('ACCELERATION') && !sensorType.includes('MAGNETIC');
+            .then((homeResponse: any) => {
+                homeID = homeResponse.id;
+                // restore login
+                AppContext.restoreLogin();
```

Comment:
The logic after this line was initially outside of the homeResponse, but we want to put it inside. The reason is because we want to wait to receive the homeResponse before fetching data, and occasionally on refresh the fetch will be performed before homeID is gotten.

Reply:
Let me add this to the list of issues for refactoring @reprimande !